### ISSUE TITLE
Coordinating applications: Community Bridge Q3-Q4 round is open for the project applications

### ISSUE THREAD

**Issue by SergeyKanzhelev** [open-telemetry/community#468]
This issue is to help coordinate applications and mentors who want to participate.

If you are active on OpenTelemetry (ideally approver or maintainer) and want to mentor somebody as part of Community Bridge, please comment on this issue.

If you have an idea for the project, please also do it in this issue before submitting to the Community Bridge repository. Ideally project should be contained and student should be able to finish it in a few weeks. Project MUST NOT be on critical path for important milestone like GA.

If you have both - project idea and ready to mentor - this is a double win!

> Greetings everyone!
> 
> CNCF is happy to announce that the Community Bridge mentorship program is open for the project applications for the Q3-Q4'2020 cycle. The full program description you may find at [the CNCF mentoring repo](https://github.com/cncf/mentoring/blob/master/communitybridge/README.md).
> 
> Project maintainers and/or potential mentors are welcome to submit their ideas via submitting a PR to GitHub [here](https://github.com/cncf/mentoring/blob/master/communitybridge/2020/q3-q4/project_ideas.md). 
>
>Please do it by September 7.
> 
> As a reminder, we had two successful mentorship rounds earlier this year ([Q1](https://www.cncf.io/blog/2020/04/15/seven-cncf-interns-graduate-from-the-2020-linux-foundation-communitybridge-program/), [Q2](https://www.cncf.io/blog/2020/08/13/21-cncf-interns-graduate-from-the-q2-2020-linux-foundation-communitybridge-program/)), with almost 30 mentees and 16 CNCF projects participating in total!

**Comment by SergeyKanzhelev** [comment]
CC:

- @open-telemetry/dotnet-approvers
- @open-telemetry/php-approvers
- @open-telemetry/javascript-approvers
- @open-telemetry/java-instrumentation-approvers
- @open-telemetry/java-approvers
- @open-telemetry/go-approvers
- @open-telemetry/ruby-approvers
- @open-telemetry/swift-approvers
- @open-telemetry/cpp-approvers
- @open-telemetry/collector-approvers
- @open-telemetry/erlang-approvers
- @open-telemetry/python-approvers
- @open-telemetry/opentelemetry-python-contrib-approvers
- @open-telemetry/rust-approvers
- @open-telemetry/docs-approvers
- @open-telemetry/docs-cn-approvers
- @open-telemetry/technical-committee

**Comment by maxgolov** [comment]
**Project**

Add exporter for **ETW (Event Tracing for Windows)** as one of the optional recommended exporters that languages may decide to implement as part of SDK. With initial implementation proposed for:
- C++
- C#

**Logical Reasoning**

Application logs, service logs, even Docker-on-Windows logs can be directed to ETW (ref. https://docs.docker.com/config/containers/logging/etwlogs/ ). Many products running on Windows (most Microsoft products, Intel products, Docker, etc.) - rely on ETW channel. Adding support for ETW for Windows platforms in SDK would allow any **cross-platform codebase** to be instrumented with OpenTelemetry instead of using Win32 API directly, e.g. replacing Docker etwlogs driver with structured ETW traces emitted using OpenTelemetry C++ SDK. Final product demo could be the Docker using OpenTelemetry C++ SDK for traces, and Docker on Windows - using OpenTelemetry C++ SDK with ETW exporter.

**Extra bonus points**

Two 'extra' pieces that can be covered in addition to the main exporter piece:

- Writing a projection layer for other languages (e.g. Java or node on Windows), that allows to dispatch traces to OpenTelemetry C++ SDK with ETW exporter. That flow would allow Java and node services / servers running on Windows hosts to route their telemetry to ETW listener agent.

- Separate effort that implements ETW Provider listener (Agent), that subsequently uses OpenTelemetry .NET SDK to dispatch all telemetry collected to OTLP exporter (or any other exporter supported by .NET SDK). Very nice and simple example how to implement a listener: https://github.com/fireeye/SilkETW

**Comment by SergeyKanzhelev** [comment]
> Add exporter for **ETW (Event Tracing for Windows)** as one of the optional recommended exporters that languages may decide to implement as part of SDK. With initial implementation proposed for:

Neat idea. Would it make more sense when we have logging APIs? Or you envision ETW exporter for Traces, Metrics would be useful by itself? Do you have end-to-end demo in mind, perhaps some consumer of ETW?

Would message pack exporter be a similar step? I can see how we can demo it with FluentBit consuming OpenTelemetry.

**Comment by maxgolov** [comment]
> Would it make more sense when we have logging APIs?

I was thinking about defining how we can express Traces and Metrics over that channel, and yes - Logging is the next step, but we do not have to wait for the formal Logging API. I think exporter for Traces and Metrics would be useful too.

> Do you have end-to-end demo in mind, perhaps some consumer of ETW?

There are several customers that I am aware of are gonna be interested in deploying this in production. We can do end-to-end demo with some exporter, for example, Application Insights, Grafana, or Google Cloud exporter. I do not have a preference of whether the Agent is gonna send the data, as long as we implement the 1st part of the flow (process to SDK to ETW), the rest of the flow from ETW listener can be bound to any destination.

> Would message pack exporter be a similar step?

It is conceptually similar - we can add that too. But - it's a bit semantically different. ETW itself allows for structured logs, and there is a number of tools (e.g. https://docs.microsoft.com/en-us/windows-hardware/test/wpt/ ) that allow to visually inspect the structured ETW logs today.

Basically there could be two tracks:
- standard structured ETW event. Pros: compatible with existing tools. Cons: flat list, requires "unflattening".
- ETW event that carries MsgPack encoded event. Pros: carries structured tree. Cons: incompatible with WPT tools.

Basically both methods could be implemented. Adding MsgPack in addition to regular structured ETW would be rather trivial (at least I can speak of C++ SDK, gonna be relatively trivial).

**Comment by SergeyKanzhelev** [comment]
https://github.com/cncf/mentoring/pull/280

**Comment by bobstrecansky** [comment]
I'd love to have some mentees work on the opentelemetry-php project, finishing some of the milestones for our releases.

**Comment by SergeyKanzhelev** [comment]
please submit a PR like this cncf/mentoring#280 asap. Deadline was yesterday

**Comment by bobstrecansky** [comment]
Thanks @SergeyKanzhelev - added [here](https://github.com/cncf/mentoring/pull/281)

**Comment by maxgolov** [comment]
@SergeyKanzhelev Thanks for adding my proposal!! Looking forward to make it happen.

**Comment by SergeyKanzhelev** [comment]
the process last time was that there will be candidates and CNCF will tell us how many spots we have for OpenTelemetry. Than we will need to match candidates and projects. I'll close this issue and will reach out when it's time for the next steps.

**Comment by tsloughter** [comment]
I don't have a proposal but I'm always open to act as a mentor if someone who is interested in Erlang/Elixir (or any BEAM language really, bindings for https://gleam.run/ would be interesting for example).

**Comment by SergeyKanzhelev** [comment]
@maxgolov @bobstrecansky can you please create an issues for the proposed projects today?

**Comment by SergeyKanzhelev** [comment]
One for FluentBit: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/31

**Comment by maxgolov** [comment]
@SergeyKanzhelev - work item for OT C++ SDK: https://github.com/open-telemetry/opentelemetry-cpp/issues/326

**Comment by SergeyKanzhelev** [comment]
Created one for php as well. https://github.com/cncf/mentoring/pull/289

**Comment by bobstrecansky** [comment]
@SergeyKanzhelev my apologies, missed the notification for this.  Thank you for opening that issue.

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev my apologies, missed the notification for this. Thank you for opening that issue.

it was a very short notice, no worries at all.

**Comment by SergeyKanzhelev** [comment]
@maxgolov @bobstrecansky can you please apply to be a mentor on Community Bridge? OpenTelemetry got a spot.

> Can you please ask them to apply directly via the Community Bridge platform ("Become a Mentor" link on the top of https://people.communitybridge.org/)?

**Comment by bobstrecansky** [comment]
Completed: https://people.communitybridge.org/mentor/9ca764eb-20b2-4d4e-a4ca-b50d0a4f0a27

**Comment by maxgolov** [comment]
@SergeyKanzhelev - I created my profile here:
https://people.communitybridge.org/mentor/314c7d13-8316-46c8-abe8-62a1d28b52b0

**Comment by SergeyKanzhelev** [comment]
I'm working as a man in the middle here. Please apply to be an OpenTelemetry mentor on the platform. I will do it tonight so will know later how exactly to do it. Shouldn't be too hard

**Comment by bobstrecansky** [comment]
I see myself listed as a Mentor for the OpenTelemetry project: https://people.communitybridge.org/project/f1275c0e-7152-4e09-8d8b-6b14598afbc3

**Comment by shivam-51** [comment]
@SergeyKanzhelev Can we not work for the ETW exporter project? I cannot find that project at CommunityBridge

**Comment by maxgolov** [comment]
@SergeyKanzhelev @bobstrecansky - would you guys please also add me as mentor? Just applied.

@shivam-51 - It's the OpenTelemetry project listed above. The task is in the opentelemetry-cpp GitHub repo:
https://github.com/open-telemetry/opentelemetry-cpp/issues/326

**Comment by SergeyKanzhelev** [comment]
CC: @idvoretskyi 

@maxgolov when did you apply for the mentorship?

**Comment by maxgolov** [comment]
> @maxgolov when did you apply for the mentorship?

@SergeyKanzhelev Yesterday.

![image](https://user-images.githubusercontent.com/34072974/94466746-c5b9f800-0176-11eb-8e96-0b778588c4bc.png)

**Comment by idvoretskyi** [comment]
@maxgolov I haven't received it for some reason. Can you please email me to 'ihor at cncf.io' with your email address and real name - I'll invite you.

Thanks.

**Comment by idvoretskyi** [comment]
@maxgolov confirmed that it's fixed.
